### PR TITLE
Block editor: use React events for shortcuts (portal bubbles & contextual)

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Builds on #32318.

Moves shortcut handlers from native events to React events for two reasons:

* Native events don't bubble through portals, which requires us to [hackily bubble events through the iframe](https://github.com/WordPress/gutenberg/commit/e54a9ec2f97e28db3293628e921e40e81cfe5f76#diff-ab17abd38e687edd56573694ae3017e105e96b319571c190e9d4705c1872bc9cR60-R104).
* The shortcuts should only work in the context of the block editor, rather than globally (on the `document`). This is important because you may have other things on the page beside the block editor, or multiple block editors. Shortcuts should only work if focus is within a certain context. Using React evens works nicely because they bubble both through the iframe and other portals such as the inspector.

Lastly, it removes some boilerplate for implementing a block editor, building `<BlockEditorKeyboardShortcuts />` into `<BlockTools />`.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by draganescu — DISMISSED** [review]
I tested this and it works in an identical way for the user. I think the reasons for the change in the PR's description are very good.

**Comment by draganescu** [comment]
Unfortunately in Safari the shortcuts are picked up by the browser even in editor context (e.g. cmd + D opens the bookmark dialog, cmd option shift t opens the profiling in Inspector).

**Comment by youknowriad** [comment]
This is a big departure in terms of APIs for the shortcuts package, I think it'd have deserved more eyes a bit.

For instance 

> The shortcuts should only work in the context of the block editor, rather than globally (on the document).

I don't think that makes sense for all shortcuts, some of the shortcuts are `edit-post` shortcuts and should work no matter where we are (like saving for instance).

I also kind of like the `useShortcut` API personally, another option could have been to return props to add to a react element if keeping the React Event approach is mandatory.

File: packages/edit-post/src/components/visual-editor/index.js [hunk 1]
```diff
@@ -192,66 +192,65 @@ export default function VisualEditor( { styles } ) {
 	}, [ isTemplateMode, themeSupportsLayout, contentSize, wideSize ] );
 
 	return (
-		<div
+		<BlockTools
```
Comment (youknowriad) [inline packages/edit-post/src/components/visual-editor/index.js:195]: For me if this needs to be a parent of the block editor, it needs to be documented properly in the block-editor package and the documentation we have to build third-party block editors.

That said, the naming is not great at all, shouldn't be `BlockEditorWrapper` or something like that?  Also, if it's a wrapper, why not just bundle it in `BlockEditorProvider`

Comment (ellatrix) [reply packages/edit-post/src/components/visual-editor/index.js:195]: Why does it need to be higher up the tree? Agreed on the documentation, I should add some.

Comment (youknowriad) [reply packages/edit-post/src/components/visual-editor/index.js:195]: I don't understand the question. My main motivation here is that it seems we're requiring two wrappers where we can require just one. Unless I'm not understanding properly both these components should wrap the editor canvas? Maybe more granularity is needed, but it's not clear from the user's perspective where to put each one of these components.

**Pull request merged**